packages/garbo/src/fights.ts

```diff
@@ -1383,6 +1391,51 @@ const freeRunFightSources = [
     },
     freeRunConstraints(true),
   ),
+  // Use free runs and free kills to get our bofa wishes
+  new FreeRunFight(
+    () =>
+      have($skill`Just the Facts`) &&
+      get("_bookOfFactsWishes") < 3 &&
+      shouldDoBofaFishing() &&
+      findFreeKill() !== null &&
+      getBestBofaWishLocation() !== null &&
+      unlock(
+        getBestBofaWishLocation(),
+        (3 - get("_bookOfFactsWishes")) * garboValue($item`pocket wish`),
+      ),
+    (runSource: ActionSource) => {
+      const best = getBestBofaWishLocation();
+      if (!best) throw `Bofa wish location should exist, but doesn't`;
+      const wishMonsters = getMonsters(best).filter(
+        (m) => itemFact(m) === $item`pocket wish`,
+      );
+      const freeKill = findFreeKill();
+      if (!freeKill) {
+        throw `We should have a free kill for bofa wishes, but don't`;
+      }
+      const freeKillMacro =
+        freeKill.macro instanceof Item
+          ? Macro.item(freeKill.macro)
+          : Macro.skill(freeKill.macro);
+      garboAdventure(
+        best,
+        Macro.if_(
+          wishMonsters.map((m) => `!monsterid ${m.id}`).join(" && "),
+          runSource.macro,
+        ).step(freeKillMacro),
```

Comment:
```suggestion
        Macro.if_(
          wishMonsters,
         freeKillMacro,
        ).step(runSource.macro),
```
just seems a little more straightforward